enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,359 @@
+required for any of those behaviors to be exposed to MicroShift admins through the
+configuration file for MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+#### 1. Set EIP through installer for the default ingress controller:
+   
+   a. The admin sets the `eip-allocations` in the installer using the install-config.yaml.
+   
+   b. The installer then sets the ingress cluster object with the `eip-allocations`.
+   
+   c. The cluster ingress operator then creates a default ingress controller by:
+
+     i.   creating an default ingress controller CR
+     
+     ii.  then scaling a deployment for the Ingress Controller CR which has the value of the `eip-allocations`.
+     
+     iii. then creating a service with service type load balancer with annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+          which uses the value from the field `eip-allocations` of ingress controller CR.
+
+#### 2. Set EIP through the Ingress Controller CRD:
+   
+   The admin sets the `eip-allocations` in the Ingress Controller CR. Then same process
+   as per point 1.c.ii and 1.c.iii is followed by the ingress operator. 
+
+#### 3. Update the existing EIP for the IngressController:
+   
+   When the admin changes the eip-allocations field in the Ingress Controller CR, the watch on the
+   `eip-allocations` field by the ingress operator scales a new deployment for the Ingress Controller CR.
+   Then step 1.c.iii is followed by the ingress operator.
+
+   
+### Risks and Mitigations
+
+Providing a right range of EIP as per AWS guidelines is an admin responsibility.
+If they are not provided the router won't get assigned an IP which can cause communication outage.
+As of now we are not able to identify any more risks which can affect other components of the EP.
+
+### Drawbacks
+
+Currently, there are not any drawbacks as such for this EP.
+
+## Open Questions
+
+1. Shall we support updating EIP in the NLB Ingress Controller ?
+
+Yes, we will support updating the EIP in the custom ingress controller.
+
+## Test Plan
+
+- This EP will be covered by unit tests.
+
+- This  EP will also be covered by end to end tests.
+  
+  The end to end test will cover the scenario where user will set an `eip-allocations` field
+  in the Ingress Controller CR and then test if the Ingress Controller
+  with service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+  has value of `eip-allocations` field from the Ingress Controller CR was created.
+
+  The end to end test will also the cover the scenario where user updates the `eip-allocations` field
+  in the Ingress Controller CR and then test if the Ingress Controller
+  with service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+  has value of `eip-allocations` field from the Ingress Controller CR was updated.
+  
```

Comment:
Is there a plan for testing when the EIP allocations are plumbed in the install-config.yaml?

Reply:
There will be unit tests in the installer.